src/com/hotmail/kalebmarc/textfighter/main/Enemy.java

```diff
@@ -24,6 +24,7 @@
     private int coinDropMax;
     private int damageMin;
     private int damageMax;
+    int damageDealt;

```

Comment:
Make this a private var, and if you need to access it from another class create a `getter` for it :)

Reply:
Alright, I did not even know something like this existed until now :D I have to say that it is much more clever. Still, I posted a question in the commit resolving this issue. I would greatly appreciate if you answered it.

Reply:
Also why can't I just make it a public variable and reference it anywhere? (This is probably common sense for you, but I do not really understand why not)

Reply:
> When getting the value, why do I have to type "Weapon.get().getDamageDealt()" instead of just "Weapon.getDamageDealt()", when the variable is not directly tied to the weapon?

Great question! This has to do with how Object-Oriented Programming works. Classes need to be initialized/created before you can use them. Think of them as templates. As I'm sure you're aware, you can have multiple instances of `Weapon`, each with its own values. The exception is `static` variables/methods, which is tied to the class itself, instead of the instance of the class. Since `getDamageDealt()` is not static, that means its tied to the instance and you can not access it directly (even though it changes with each round, it technically is still a property of `Weapon`).

`Weapon.get()` is a static method for getting the currently "held" weapon. When calling this, you are actually getting the instance of `Weapon`, and not `Weapon` itself

Let me know if this is still confusing, I can write a better example if needed!


> Also why can't I just make it a public variable and reference it anywhere? 
There's nothing "illegal" about doing it this way, it works just fine, a lot of projects will do it this way just because its easier. Though, it is common (and good) practice to not allow direct access to variables. There are plenty of benefits, including:
- Security. It can help prevent accidental changes to the variable (you might only want to allow reading the variable, not changing it)
- You can add validation if needed (using a setter to make sure code isn't trying to set it to a value it shouldn't be)
[here's a more technical answer](https://stackoverflow.com/a/1568230/3291305)

[check this out as well](https://www.w3schools.com/java/java_encapsulation.asp)

Note that most of these things usually aren't a problem in smaller projects like this, but become much more important in large projects with dozens of people sharing the same codebase; however, it's always smart to plan ahead and follow good coding styles!

Again, let me know if that still doesn't make sense and I can get better examples for you

Reply:
> Thank you so much for the answers, they have really helped me a lot.
No problem! I was in your situation a few years ago, so I definitely understand what it's like; I'm more than happy to help out where I can!

Feel free to reach out whenever, even with questions about your own projects (either code questions, or high-level concepts), or if you just want some code reviews on something! My email is available on my Github profile :)